### DISCUSSION TITLE
password in clear text in config file

### DISCUSSION THREAD

**Discussion by mrsteve0924** [any1/wayvnc#272] [Q&A]
is there a way to use an encrypted password file like tigervnc has?

**Comment by any1** [comment]
There are multiple authentication methods supported. The password would have to be hashed separately for all of them.

**Reply by mrsteve0924** [reply]
i dont see anything in the man pages or README about password files. 

in tigervnc if i issue the command vncpasswd it will ask for a new password and store an obfuscated version of it to a password file. i can then pass it as an option to  the vncserver command.

**Reply by any1** [reply]
I didn't say that password files were supported.

From `vncpasswd(1)`:

> The password must be at least six characters long (unless the -f command-line
>        option is used-- see below), and only the first eight characters are signifi‐
>        cant.  **Note that the stored password is not encrypted securely -  anyone  who
>        has  access  to  this file can trivially find out the plain-text password**, so
>        vncpasswd always sets appropriate permissions (read and  write  only  by  the
>        owner.)   However,  when accessing a VNC desktop, a challenge-response mecha‐
>        nism is used over the wire making it hard for anyone to  crack  the  password
>        simply by snooping on the network.

So, what's the point?

**Comment by mrsteve0924** [comment]
well that is interesting.  i dont know why they say "can trivially find out the plain text password" , when i open my password file, it is just a mess of random characters.  i tried searching for some tools to read this file as plain text, found one but it was for Windows. 

so it is what it is  i guess. thanks for the help.

**Reply by Consolatis** [reply]
Here you go: https://github.com/TigerVNC/tigervnc/blob/master/common/rfb/obfuscate.cxx

So its d3des with a publicly known key.
Basically the only reason I can imagine this exists is to address the 'somebody-looking-over-the-shoulder' factor.

**Reply by mrsteve0924** [reply]
yes as i am reading more about it, their password encryption does seem pretty useless.  even their source code has a warning about it
// XXX not thread-safe, because d3des isn't - do we need to worry about this?
